File: business-integrator/src/main/java/groupB/newbankV5/businessIntegrator/controllers/dto/MerchantDto.java
package groupB.newbankV5.businessIntegrator.controllers.dto;

import groupB.newbankV5.businessIntegrator.entities.BankAccount;
import groupB.newbankV5.businessIntegrator.entities.Merchant;

public class MerchantDto {
    private Long id;
    private String name;
    private String email;
    private BankAccount bankAccount;

    public Long getId() {
        return id;
    }

    public void setId(Long id) {
        this.id = id;
    }

    public String getName() {
        return name;
    }

    public void setName(String name) {
        this.name = name;
    }

    public String getEmail() {
        return email;
    }

    public void setEmail(String email) {
        this.email = email;
    }

    public BankAccount getBankAccount() {
        return bankAccount;
    }

    public void setBankAccount(BankAccount bankAccount) {
        this.bankAccount = bankAccount;
    }



    public static MerchantDto merchantDtoFactory(Merchant merchant) {
        MerchantDto merchantDto = new MerchantDto();
        merchantDto.setId(merchant.getId());
        merchantDto.setName(merchant.getName());
        merchantDto.setEmail(merchant.getEmail());
        merchantDto.setBankAccount(merchant.getBankAccount());
        return merchantDto;
    }
}

File: web-sdk/src/sdk/dto/merchant.dto.ts
import { BankAccountDTO } from './bank-account.dto';
import { ApplicationDto } from './application.dto';
export class MerchantDTO {
  id: number;
  name: string;
  email: string;
  bankAccount: BankAccountDTO;
  application: ApplicationDto;

  constructor(
    id: number,
    name: string,
    email: string,
    bankAccount: BankAccountDTO,
    application: ApplicationDto,
  ) {
    this.id = id;
    this.name = name;
    this.email = email;
    this.bankAccount = bankAccount;
    this.application = application;
  }
}
